Tribler/Core/Modules/restapi/shutdown_endpoint.py

```diff
@@ -0,0 +1,53 @@
+import os
+import json
+
+from twisted.web import resource
+from twisted.internet import reactor
+
+from Tribler.Core.Modules.process_checker import ProcessChecker
+
+
+class ShutdownEndpoint(resource.Resource):
+    """
+    With this endpoint you can trigger a Tribler shutdown.
+    """
+
+    def __init__(self, session):
+        resource.Resource.__init__(self)
+        self.session = session
+        self._stopping = False
+        self.process_checker = ProcessChecker()
+
+    def render_PUT(self, request):
+        """
+        .. http:put:: /shutdown
+
+        A PUT request to this endpoint triggers a Tribler shutdown.
+
+            **Example request**:
+
+            .. sourcecode:: none
+
+                curl -X PUT http://localhost:8085/shutdown
+
+            **Example response**:
+
+            .. sourcecode:: javascript
+
+                {
+                    "shutdown": True,
+                    "gracetime": 2.0
+                }
+        """
+        gracetime = 2.0
+
+        def shutdown_process(_, code=1):
+            reactor.addSystemEventTrigger('after', 'shutdown', os._exit, code)
+            reactor.stop()
+            self.process_checker.remove_lock_file()
+
+        if not self._stopping:
+            self._stopping = True
```

Comment:
@whirm @devos50  Is the endpoint instantiated for every request? If so, this ```if _stopping``` does not help.